### ISSUE TITLE
Fashion loadouts cannot remove shader from armor with non-default singleInitialItemHash shader

### ISSUE THREAD

**Issue by robojumper** [DestinyItemManager/DIM#7720]
### DIM Version

Version 6.99.1.1001183 (beta), built on 10.1.2022, 09:41:11

### Browser Details

Firefox 95.0.2

### OS Details

Windows 10

### Describe the bug

![grafik](https://user-images.githubusercontent.com/14299449/148797094-ef5eb071-fbe4-4048-9111-d8f1d99644db.png)

I think the problem is that [Annealed Shaper Robes'](https://data.destinysets.com/i/InventoryItem:2011267747) shader socket has a `singleInitialItemHash` referencing the [New Age Black Armory](https://data.destinysets.com/i/InventoryItem:3650581584) shader, so loadout application doesn't consider the default shader available. The "default shader" plug isn't enabled by the shader plug set, but rather by the item/inventory response, so loadout application and fashion drawer don't consider it available.

![grafik](https://user-images.githubusercontent.com/14299449/148799669-0b3f6d85-3bf7-4b7f-b906-15245d8816b4.png)

Here the default shader is selected but dimmed out because it's not considered available.

### Logs

```shell
[loadout] Apply loadout Fashion1 to Human Warlock 
[loadout mods] Mods to apply  Array []
[loadout mods] all mods are already there. loadout already applied
[loadout mods] Mods applied 2 2
[loadout] Apply loadout Fashion2 to Human Warlock
[loadout mods] Mods to apply  Array []
[loadout mods] Applying Array [] to Verity's Brow
[loadout mods] Applying Array [] to Gloves of the Hezen Lords
[loadout mods] Applying Array [ {…} ] to Annealed Shaper Robes
[loadout mods] Applying Array [] to Tread of the Hezen Lords
[loadout mods] Applying Array [] to Phoenix's Fire
[loadout mods] Mods applied 1 2
[loadout] failed loadout 
Object { phase: 8, equipNotPossible: false, itemStates: {…}, socketOverrideStates: {}, modStates: (2) […] }
equipNotPossible: false
itemStates: Object { 6917529503504855333: {…} }
6917529503504855333: Object { item: {…}, equip: true, state: 1 }
equip: true
item: Object { owner: "2305843009404625776", destinyVersion: 2, hash: 2011267747, … }
state: 1
<prototype>: Object { … }
<prototype>: Object { … }
​modStates: Array [ {…}, {…} ]
0: Object { modHash: 702981643, state: 2, error: undefined }
1: Object { modHash: 4248210736, state: 1, error: DimError }
error: DimError: Mod didn't fit on your current armor
modHash: 4248210736
state: 1
<prototype>: Object { … }
length: 2
<prototype>: Array []
phase: 8
socketOverrideStates: Object {  }
<prototype>: Object { … }
<prototype>: Object { … }
```

**Comment by bhollis** [comment]
That's gonna be a tricky one - I suppose we'll need to change all our notions of "default plug hash" (e.g. `getDefaultPlugHash`, `isAssigningToDefault`) to instead consider whether we're *clearing* the socket, and we'll probably need to manually add the standard "default plug" to those sockets. Does that sound right?

**Comment by robojumper** [comment]
Some notion of what "clear" is for each socket type could probably be useful but that info isn't easy to guess and probably not necessary for this shaders issue, because at this stage we *should* have item instance data from the inventory response and the "clear" shader should be present in the API `reusablePlugs` (and added to our DIM `plugOptions`). I think the thing that fails immediately here is https://github.com/DestinyItemManager/DIM/blob/d47c6fdff32b5637706c3930fa885bf1fd45dc6c/src/app/utils/socket-utils.ts#L152-L164, because shaders use a combination of item plugs (for the and profile/character plug sets.

**Comment by robojumper** [comment]
Side note, that even the `singleInitialItemHash` isn't always enabled can be seen in Masterworks, e.g. the plug picker (`SocketDetails`) thinks a Range 1 MW could be inserted into my Hothead. There's in general a bit of a conflict between what the plug picker thinks and what loadout application thinks and it'd probably make sense to teach the plug picker to think like loadouts, which could catch these issues more reliably.

**Comment by robojumper** [comment]
I briefly took another look at this and realized a holistic fix would require some deeper changes to how we handle the API -> DIM translation for sockets and plugs, because it looks like we lose the empty shader as an option pretty early... https://github.com/DestinyItemManager/DIM/blob/9a04cd78d3bbf2334021ac0729988bfda38e5f3a/src/app/inventory/store/sockets.ts#L444-L446

**Comment by robojumper** [comment]
I tried to imagine what a fix could look like and realized we lose the empty shader as an option even earlier when trying to optimize the size of the stored manifest: https://github.com/DestinyItemManager/DIM/blob/903b1dec0bb80d4883e5eec658e4a8c6c9b68b5f/src/app/manifest/manifest-service-json.ts#L57-L64

**Comment by bhollis** [comment]
Good catch. We could do something smarter there - originally those got cleaned out because it's a lot of redundant data since plugsets became a thing.
